tests/framework/cli/test_jupyter.py

```diff
@@ -22,209 +22,147 @@ def mocked_logging(mocker):
+        kernel_name = f"kedro_{fake_metadata.package_name}"
+        display_name = f"Kedro ({fake_metadata.package_name})"
+        create_kernel_mock.assert_called_once_with(kernel_name, display_name)
         python_call_mock.assert_called_once_with(
-            *default_jupyter_options("lab", all_kernels=True)
-        )
-
-    @pytest.mark.parametrize("help_flag", ["-h", "--help"])
-    def test_help(
-        self, help_flag, fake_project_cli, fake_ipython_message, fake_metadata
-    ):
-        result = CliRunner().invoke(
-            fake_project_cli, ["jupyter", "lab", help_flag], obj=fake_metadata
+            "jupyter",
+            [
+                "lab",
+                f"--MultiKernelManager.default_kernel_name={kernel_name}",
+                "--random-arg",
+                "value",
+            ],
         )
-        assert not result.exit_code, result.stdout
-        fake_ipython_message.assert_not_called()
 
-    @pytest.mark.parametrize("env_flag", ["--env", "-e"])
-    def test_env(self, env_flag, fake_project_cli, python_call_mock, fake_metadata):
+    @pytest.mark.parametrize("env_flag,env", [("--env", "base"), ("-e", "local")])
+    def test_env(self, env_flag, env, fake_project_cli, fake_metadata, mocker):
         """This tests passing an environment variable to the jupyter subprocess."""
+        mock_environ = mocker.patch.dict("kedro.framework.cli.jupyter.os.environ")
         result = CliRunner().invoke(
             fake_project_cli,
-            ["jupyter", "lab", env_flag, "base"],
+            ["jupyter", "lab", env_flag, env],
             obj=fake_metadata,
         )
-        assert not result.exit_code
-
-        args, kwargs = python_call_mock.call_args
-        assert args == default_jupyter_options("lab")
-        assert "env" in kwargs
-        assert kwargs["env"]["KEDRO_ENV"] == "base"
+        assert not result.exit_code, result.stdout
+        assert mock_environ["KEDRO_ENV"] == env
 
-    def test_fail_no_jupyter_core(self, fake_project_cli, mocker):
-        mocker.patch.dict("sys.modules", {"jupyter_core": None})
+    def test_fail_no_jupyter(self, fake_project_cli, mocker):
+        mocker.patch.dict("sys.modules", {"jupyterlab": None})
         result = CliRunner().invoke(fake_project_cli, ["jupyter", "lab"])
 
         assert result.exit_code
         error = (
-            "Module `jupyter_core` not found. Make sure to install required project "
+            "Module `jupyterlab` not found. Make sure to install required project "
             "dependencies by running the `pip install -r src/requirements.txt` command first."
         )
         assert error in result.output
 
 
+@pytest.fixture
+def cleanup_kernel():
+    yield
+    if "my_kernel_name" in find_kernel_specs():
+        KernelSpecManager().remove_kernel_spec("my_kernel_name")
+
+
+@pytest.mark.usefixtures("cleanup_kernel")
+class TestCreateKernel:
+    def test_create_new_kernel(self):
```

Comment:
I decided to do this as a sort of e2e test that checks what the actual result of calling `_create_kernel` is rather than mocking out all those calls. This means its not tied to a particular implementation and will be easy to modify the tests if and when this becomes its own `kedro jupyter-init` command.